#! /bin/sh /usr/share/dpatch/dpatch-run
## 01-cf.data.debian.dpatch by Luigi Gangitano <luigi@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Default configuration file for debian

@DPATCH@
diff -urNad squid3~/src/cf.data.pre squid3/src/cf.data.pre
--- squid3~/src/cf.data.pre	2007-12-17 11:35:55.000000000 +0100
+++ squid3/src/cf.data.pre	2007-12-17 15:10:26.000000000 +0100
@@ -107,7 +107,7 @@
 	If you want to use the traditional NCSA proxy authentication, set
 	this line to something like
 
-	auth_param basic program @DEFAULT_PREFIX@/libexec/ncsa_auth @DEFAULT_PREFIX@/etc/passwd
+	auth_param basic program @DEFAULT_PREFIX@/lib/squid3/ncsa_auth @DEFAULT_PREFIX@/etc/passwd
 
 	"children" numberofchildren
 	The number of authenticator processes to spawn. If you start too few
@@ -169,7 +169,7 @@
 	If you want to use a digest authenticator, set this line to
 	something like
 
-	auth_param digest program @DEFAULT_PREFIX@/bin/digest_auth_pw @DEFAULT_PREFIX@/etc/digpass
+	auth_param digest program @DEFAULT_PREFIX@/lib/squid3/digest_auth_pw @DEFAULT_PREFIX@/etc/digpass
 
 	"children" numberofchildren
 	The number of authenticator processes to spawn (no default).
@@ -225,7 +225,7 @@
 	of type proxy_auth.  By default, the NTLM authenticator_program
 	is not used.
 
-	auth_param ntlm program @DEFAULT_PREFIX@/bin/ntlm_auth
+	auth_param ntlm program @DEFAULT_PREFIX@/lib/squid3/ntlm_auth
 
 	"children" numberofchildren
 	The number of authenticator processes to spawn (no default).
@@ -260,7 +260,7 @@
 	The only supported program for this role is the ntlm_auth
 	program distributed as part of Samba, version 4 or later.
 
-	auth_param negotiate program @DEFAULT_PREFIX@/bin/ntlm_auth --helper-protocol=gss-spnego
+	auth_param negotiate program @DEFAULT_PREFIX@/lib/squid3/ntlm_auth --helper-protocol=gss-spnego
 
 	"children" numberofchildren
 	The number of authenticator processes to spawn (no default).
@@ -607,9 +607,9 @@
 # Example rule allowing access from your local networks.
 # Adapt to list your (internal) IP networks from where browsing
 # should be allowed
-acl localnet src 10.0.0.0/8	# RFC1918 possible internal network
-acl localnet src 172.16.0.0/12	# RFC1918 possible internal network
-acl localnet src 192.168.0.0/16	# RFC1918 possible internal network
+#acl localnet src 10.0.0.0/8	# RFC1918 possible internal network
+#acl localnet src 172.16.0.0/12	# RFC1918 possible internal network
+#acl localnet src 192.168.0.0/16	# RFC1918 possible internal network
 #
 acl SSL_ports port 443
 acl Safe_ports port 80		# http
@@ -670,7 +670,8 @@
 # Example rule allowing access from your local networks.
 # Adapt localnet in the ACL section to list your (internal) IP networks
 # from where browsing should be allowed
-http_access allow localnet
+#http_access allow localnet
+http_access allow localhost
 
 # And finally deny all other access to this proxy
 http_access deny all
@@ -709,7 +710,7 @@
 
 NOCOMMENT_START
 #Allow ICP queries from local networks only
-icp_access allow localnet
+#icp_access allow localnet
 icp_access deny all
 NOCOMMENT_END
 DOC_END
@@ -734,7 +735,7 @@
 
 NOCOMMENT_START
 #Allow HTCP queries from local networks only
-htcp_access allow localnet
+#htcp_access allow localnet
 htcp_access deny all
 NOCOMMENT_END
 DOC_END
@@ -2080,7 +2081,7 @@
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
 	Specifies the number of logfile rotations to make when you
@@ -2096,6 +2097,9 @@
 	purposes, so -k rotate uses another signal.  It is best to get
 	in the habit of using 'squid -k rotate' instead of 'kill -USR1
 	<pid>'.
+	
+	Note2, for Debian/Linux the default of logfile_rotate is
+	zero, since it includes external logfile-rotation methods.
 DOC_END
 
 NAME: emulate_httpd_log
@@ -3274,7 +3278,7 @@
 NAME: visible_hostname
 TYPE: string
 LOC: Config.visibleHostname
-DEFAULT: none
+DEFAULT: localhost
 DOC_START
 	If you want to present a special hostname in error messages, etc,
 	define this.  Otherwise, the return value of gethostname()
